modules/contact/src/MechanicalContactConstraint.C

```diff
@@ -226,6 +233,9 @@ MechanicalContactConstraint::timestepSetup()
       updateAugmentedLagrangianMultiplier(true);
 
     _update_stateful_data = false;
+
+    if (_contact_linesearch)
+      _contact_linesearch->nl_its() = 0;
```

Comment:
OK - here you are writing to that one member that's exposed, but this breaks encapsulation. You should be able to set this directly in the linesearch object.